elastic/kibana
# [Actions] Adjust axios payload for APIs that do not want `data` sent

## Problem: Cases Webhook Connector throwing vague error

A customer reported issues while using the Webhook - Case Management connector (`cases_webook`). This is the error they saw:

<img width="1489" alt="Screenshot 2023-09-05 at 12 36 49 PM" src="https://github.com/elastic/kibana/assets/6935300/62860d5a-f3cc-4b56-8be0-86fd9250458d">

After gaining access to an instance of TheHive5, I was able to narrow down the issue to the GET incident call. When I removed the data object from being sent as `{}` when undefined, the error went away and the call succeeded. I used `curl` against the Hive5 API directly. As suspected, when a data object is included the API throws: `{"type":"NotFoundError","message":"/api/v1/case/~122917056"}`. It seems likely axios swallowed that error. 

## Solution: Removing `data` when undefined
As mentioned above, I am removing the axios data argument instead of being sent as `{}` when undefined. A GET request doesn't usually transmit data in any other way besides passing an encoded string in the URL. I think it makes sense to not send data when it is undefined vs sending an empty object as it can break some 3rd party APIs as we are seeing with Hive5

## To test

1. Use these TheHive5 [credentials](https://p.elstc.co/paste/k8RMW6O+#76txL8mcohYKnH1jR8nCopq+bDd9fNHV9LWd8P+azzj). to create a cases webhook connector.
2. Run the test, post a sample case. 
3. Confirm you see a successful post: 
     <img width="698" alt="Screenshot 2023-09-05 at 12 58 29 PM" src="https://github.com/elastic/kibana/assets/6935300/b724b3dc-7a37-49c6-bae0-348c2e8ecf40">

4. Extra confirm by logging into the hive with the above credentials and see the case you created there
